From bae1baa18036c4b401325fe4c2e32e6892576cbe Mon Sep 17 00:00:00 2001
From: Roman Li <Roman.Li@amd.com>
Date: Tue, 19 Jul 2016 15:01:42 -0400
Subject: [PATCH 0658/1722] drm/amd/dal: Hot-Plug crash.

Don't call dm_handle_hpd_rx_irq until link is
established.

Signed-off-by: Roman Li <Roman.Li@amd.com>
Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
index f7714b1..82fa2a9 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
@@ -955,8 +955,8 @@ static void handle_hpd_rx_irq(void *param)
 			drm_kms_helper_hotplug_event(dev);
 		}
 	}
-
-	dm_handle_hpd_rx_irq(aconnector);
+	if (aconnector->dc_link->cur_link_settings.lane_count != LANE_COUNT_UNKNOWN)
+		dm_handle_hpd_rx_irq(aconnector);
 }
 
 static void register_hpd_handlers(struct amdgpu_device *adev)
-- 
2.7.4

